src/Node.hs

```diff
@@ -98,7 +100,7 @@ listenerMessageName (ListenerActionConversation f) =
                 -> Proxy rcv
                 -> MessageName
         msgName _ = messageName
-    in  msgName (f undefined) Proxy
+    in  msgName (f undefined undefined) Proxy
```

Comment:
I think we can avoid the scary `undefined`s and do this using scoped type variables. Give a type signature for `f`, i.e. `LL.NodeId -> SendActions packing m -> ConversationActions snd rcv m -> m ()`, so that `rcv` is in scope. Then use `messageName (Proxy :: Proxy rcv)`.